src/net.cpp

```diff
@@ -2655,8 +2655,8 @@ CNode::CNode(NodeId idIn, ServiceFlags nLocalServicesIn, int nMyStartingHeightIn
     addrKnown(5000, 0.001),
     // Don't relay addr messages to peers that we connect to as block-relay-only
     // peers (to prevent adversaries from inferring these links from addr
-    // traffic).
-    m_addr_relay_peer(!block_relay_only),
+    // traffic) or SPV clients (because they don't store or relay addresses).
+    m_addr_relay_peer(!block_relay_only && MayHaveUsefulAddressDB(nServices)),
```

Comment:
I believe that since this is the constructor, `nServices` is always `NODE_NONE` here

Reply:
Yeah, will fix soon.